widgets/contact/contact.php

```diff
@@ -697,19 +721,24 @@ function modify_instance( $instance ) {
 	}
 
 	function get_template_variables( $instance, $args ) {
-		$vars = array();
-
 		unset( $instance['title'] );
 		unset( $instance['display_title'] );
 		unset( $instance['design'] );
 		unset( $instance['panels_info'] );
 
 		// Include '_sow_form_id' in generation of 'instance_hash' to allow multiple instances of the same form on a page.
-		$vars['instance_hash'] = md5( serialize( $instance ) );
-
+		$instance_hash = md5( serialize( $instance ) );
 		unset( $instance['_sow_form_id'] );
 
-		return $vars;
+		if ( ! empty( $instance['settings']['submit_id'] ) ) {
+			$submit_attributes['id'] = $instance['settings']['submit_id'];
```

Comment:
I don't see `$submit_attributes` being declared or initialized anywhere? Is it necessary to make this an `array`? It seems like it only ever contains `'id'`,

Reply:
That's correct. I based this purely on how we handle other buttons in SiteOrigin Widgets Bundle - they all have an option of outputting additional values based on the contents of `$submit_attributes`. This allows other developers to easily add additional attributes to the button without having to override the template.

If you would rather not do this for this button, that's fine - just say the word and I'll remove it.

Reply:
Ok that's cool. Then we'd still need `$submit_attributes` to be declared and initialized before using it. And to allow third party devs to alter it's contents you'd need to add a filter.